modules/kubeflow-mlflow/tests/test_deployment.py

```diff
@@ -0,0 +1,131 @@
+    return client
+
+
+class TestCharm:
+
+    @pytest.mark.dependency()
+    async def test_apply_terraform_solution(self):
+        """Initialize and apply the kubeflow-mlflow Terraform solution module."""
+        subprocess.run(["terraform", "init"], check=True)
+        # Due to https://github.com/canonical/mysql-k8s-operator/issues/504,
+        # we need to pin instances of mysql-k8s (using the one from 8.0/beta).
+        # Otherwise, mysql will error out with COS configuration enabled.
+        subprocess.run(
+            [
+                "terraform",
+                "apply",
+                "-var",
+                "cos_configuration=true",
+                "-var",
+                "kfp_db_revision=203",
+                "-var",
+                "katib_db_revision=203",
+                "-var",
+                "mlflow_mysql_revision=203",
+                "-auto-approve",
+            ],
+            check=True,
+        )
+
+    @pytest.mark.dependency(depends=["TestCharm::test_apply_terraform_solution"])
+    async def test_assert_deployment(self, ops_test: OpsTest, lightkube_client):
+        """
+        Wait for the applications to become active and idle and verify its public URL access.
+        """
+
+        # We cannot wait the model because grafana-agent-k8s
+        # remains blocked until it's related to one of the COS charms
+        # Thus we define all charms deployed by the solution except this.
+        await ops_test.model.wait_for_idle(
+            apps=[
+                "admission-webhook",
+                "argo-controller",
+                "dex-auth",
+                "envoy",
+                "istio-ingressgateway",
+                "istio-pilot",
+                "jupyter-controller",
+                "jupyter-ui",
+                "katib-controller",
+                "katib-db",
+                "katib-db-manager",
+                "katib-ui",
+                "kfp-api",
+                "kfp-db",
+                "kfp-metadata-writer",
+                "kfp-persistence",
+                "kfp-profile-controller",
+                "kfp-schedwf",
+                "kfp-ui",
+                "kfp-viewer",
+                "kfp-viz",
+                "knative-eventing",
+                "knative-operator",
+                "knative-serving",
+                "kserve-controller",
+                "kubeflow-dashboard",
+                "kubeflow-profiles",
+                "kubeflow-roles",
+                "kubeflow-volumes",
+                "metacontroller-operator",
+                "minio",
+                "mlflow-minio",
+                "mlflow-mysql",
+                "mlflow-server",
+                "mlmd",
+                "oidc-gatekeeper",
+                "pvcviewer-operator",
+                "resource-dispatcher",
+                "tensorboard-controller",
+                "tensorboards-web-app",
+                "training-operator",
+            ],
```

Comment:
suggestion: can we get the list of deployed charms dynamically and remove `grafana-agent-k8s` from there? It's likely that we miss something in this list if any changes were to occur to the deployed charms

Reply:
Very good idea actually, updated!